src/impl/epoll/external_commit_helper.cpp

```diff
@@ -131,16 +129,51 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
         throw std::system_error(errno, std::system_category());
     }
 
+    if (!s_daemon_thread) {
+        s_daemon_thread = std::make_unique<DaemonThread>();
+    }
+    // Lock is inside add_commit_helper. The fifo and thread creation is fine since RealmCoordinator has lock when
+    // calling this.
```

Comment:
Relying on details of how the caller performs locking seems fragile. My inclination would be to change the daemon thread to not be destroyed after it's created, and to rely on the thread-safe initialization of function-level statics to guard the initialization of the thread object. This would remove all reliance on external synchronization and simplify this code.